src/Themes/ui/Theme.tsx

```diff
@@ -185,10 +185,20 @@ export function refreshTheme(): void {
             "&:hover": {
               backgroundColor: Settings.theme.backgroundsecondary,
             },
-
             borderRadius: 0,
           },
         },
+        variants: [
+          {
+            props: { variant: "maxed" },
```

Comment:
So, I'm not completely against this, but... why are maxed things special in this way? What makes a maxed button unique, such that it should have a different visual affordance beyond simply being disabled?

If we are going to do this, I think it implies a larger effort that might either be outside the scope of this PR, or greatly expand the scope of this PR, because there are a *lot* of things you can max in this game, and presumably they all would deserve the same treatment?

Reply:
The difference is you can eventually (by getting enough money) go from disabled to not, but nothing[1] will cause a button to go from maxed to not. I think making this easy to see at a glance is worthwhile.


[1] There are exceptions, such as `NS.deleteServer`.

Reply:
Well, installing augmentations will cause these buttons to stop being maxed, but I could see considering that in a different "category" of actions.

But I'm also interested in the 2nd part. By that logic, it *seems* like this styling should apply to almost all maxable buttons - upgrades in hashnet, corp, etc. I'm on the fence about whether styling these differently from disabled is helpful, but I think styling them differently from all the other things that we can max doesn't seem to make sense.

Reply:
So, the reason I went with a distinct look (at least from disabled, if not also from not-disabled), is that that was the existing behaviour of the cores and ram buttons.

When I had maxed out cores and not ram, it would show
- the core button as all green
- the ram button 
  - green with yellow money if I could afford something
  - turquoise if I couldn't

That makes it easy to see at a glance what I could buy, what I could eventually afford, and what I had already maxed out.

That would probably make sense for other things, but I've not run into them often enough to notice the behaviour one way or another. Except for the servers, which is what prompted this PR.

Reply:
I just noticed that the same issue applies to the "TOR router" button - it also appears as interactable when the TOR router is purchased, probably for the same underlying reason.

What about this as an option?
* Able to upgrade and have enough money: Green with yellow money, same as now
* Unable to upgrade due to not enough money: All turquoise, same as now
* Unable to upgrade due to all upgrades purchased: Turquoise label, quantity ("MAX" or "MAX SERVERS") in green

This would satisfy the desire to have the three options be visually distinct. It can also be implemented without needing a new variant for the Button. And it makes sense situationally; the label is still visually disabled, but with an additional flair to indicate that it's different from "normal" disabled. This might, or might not, make sense to apply to other buttons.